### DISCUSSION TITLE
petsc install failure during firedrake install

### DISCUSSION THREAD

**Discussion by hirani** [firedrakeproject/firedrake#2758] [Firedrake support] [answered]
I'm trying to install firedrake on a Mac running MacOS Ventura. The installation of petsc was not successful. Here is the relevant error output:

> Installing petsc/
Downloading Eigen from 'https://github.com/eigenteam/eigen-git-mirror/archive/3.3.3.tar.gz' to 'eigen-3.3.3.tgz'
Checking Eigen tarball integrity
Eigen tarball hash valid
Depending on your platform, PETSc may take an hour or more to build!
Traceback (most recent call last):
  File "/Users/hirani/firedrake-install", line 1854, in <module>
    install("petsc/")
  File "/Users/hirani/firedrake-install", line 1022, in install
    build_and_install_petsc()
  File "/Users/hirani/firedrake-install", line 1161, in build_and_install_petsc
    check_call([python, "./configure", "PETSC_DIR={}".format(petsc_dir), "PETSC_ARCH={}".format(petsc_arch)] + petsc_options)
  File "/Users/hirani/firedrake-install", line 655, in check_call
    log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT, env=os.environ).decode())
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/subprocess.py", line 421, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/Users/hirani/firedrake/bin/python', './configure', 'PETSC_DIR=/Users/hirani/firedrake/src/petsc', 'PETSC_ARCH=default', '--with-debugging=0', '--download-ptscotch', '--download-pastix', '--download-hwloc', '--download-hwloc-configure-arguments=--disable-opencl', '--download-mumps', '--download-suitesparse', '--with-zlib', '--CFLAGS=-I/opt/homebrew/opt/openblas/include -Wno-implicit-function-declaration', '--with-shared-libraries=1', '--download-netcdf', '--download-bison', '--download-mpich', '--download-hdf5', '--download-hypre', '--LDFLAGS=-Wl,-rpath,/opt/homebrew/opt/openblas/lib -L/opt/homebrew/opt/openblas/lib', '--download-pnetcdf', '--download-ml', '--with-fortran-bindings=0', '--download-eigen=/Users/hirani/firedrake/src/eigen-3.3.3.tgz ', '--with-x=0', '--download-mpich-configure-arguments=--disable-opencl', '--download-chaco', '--with-blaslapack-dir=/opt/homebrew/opt/openblas', '--download-scalapack', '--download-metis', '--with-c2html=0', '--download-superlu_dist']'
returned non-zero exit status 1.

Here are the log files:
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/10679588/firedrake-install.log)
[configure.log](https://github.com/firedrakeproject/firedrake/files/10679597/configure.log)

**Comment by connorjward** [comment, marked as answer]
Please try following the steps described in https://github.com/firedrakeproject/firedrake/issues/2689#issuecomment-1352884934

**Reply by hirani** [reply]
Thanks.  That worked. As suggested in the answer, in `firedrake-install` file I changed:
`with environment(MACOSX_DEPLOYMENT_TARGET="10.15")`
to
`with environment(MACOSX_DEPLOYMENT_TARGET="11")`
and firedrake installation worked.
